tgui/packages/tgui/interfaces/NtosMain.js

```diff
@@ -40,23 +45,32 @@ export const NtosMain = (props, context) => {
             </Button>
           </Section>
         )}
-        {!!cardholder && (
+        {!!(cardholder && show_imprint) && (
           <Section
             title="User Login"
             buttons={(
-              <Button
-                icon="eject"
-                content="Eject ID"
-                disabled={!login.IDName}
-                onClick={() => act('PC_Eject_Disk', { name: "ID" })}
-              />
+              <Box>
+                <Button
+                  icon="eject"
+                  content="Eject ID"
+                  disabled={!proposed_login.IDName}
+                  onClick={() => act('PC_Eject_Disk', { name: "ID" })}
+                />
+                <Button
+                  icon="dna"
+                  content="Imprint ID"
+                  // eslint-disable-next-line max-len
+                  disabled={!proposed_login.IDName || ((proposed_login.IDName === login.IDName) && (proposed_login.IDJob === login.IDJob))}
```

Comment:
Just 'splode it into more than one line, rather than disabling the eslint, it's there for a reason.